### PR TITLE
Winter16 jetID update to include lepton veto 

### PR DESCRIPTION
**Pull request by knash**
This is a requested change to the jetID code to support the Winter16 TIGHTLEPVETO operating point.  

Additionally, a type-mismatch issue was found when setting custom user parameters.  This issue would have resulted in an error in the case where a user wanted to supply a custom operating point for a specific JetID variable.  This is a yet to be used functionality and does not affect the JetID value in any way, but in theory it should be useable for internal tests.

### CODE REVIEW

File: PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h [hunk 1]
```diff
@@ -416,10 +419,10 @@ class PFJetIDSelectionFunctor : public Selector<pat::Jet>  {
...
       }
 
-   if( (version_ != WINTER17 && version_ != WINTER17PUPPI) && quality_ ==  TIGHTLEPVETO){
-      edm::LogWarning("BadJetIDVersion") << "JetID version does not support the TIGHTLEPVETO operating point -- defaulting to TIGHT";
-      quality_ = TIGHT;
-      }
+   //if( (version_ != WINTER17 && version_ != WINTER17PUPPI) && quality_ ==  TIGHTLEPVETO){
+   //   edm::LogWarning("BadJetIDVersion") << "JetID version does not support the TIGHTLEPVETO operating point -- defaulting to TIGHT";
+   //   quality_ = TIGHT;
+   //   }
```
Comment (slava77) [inline PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h]: is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

File: PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h [hunk 2]
```diff
@@ -480,11 +483,11 @@ class PFJetIDSelectionFunctor : public Selector<pat::Jet>  {
     }else if ( quality_ == TIGHTLEPVETO ) {
       set("CHF", 0.0);
       set("NHF", 0.9);
-      set("CEF", 0.8);
       set("NEF", 0.9);
       set("NCH", 0);
       set("nConstituents", 1);
       if(version_ == WINTER17){
+        set("CEF", 0.8);
 	set("NEF_EC_L",0.02);
```
Comment (slava77) [inline PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h]: here and below the indentation inside the ```if``` is inconsistent.

File: PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h [hunk 3]
```diff
@@ -494,13 +497,23 @@ class PFJetIDSelectionFunctor : public Selector<pat::Jet>  {
         set("MUF", 0.8);
       }
       if(version_ == WINTER17PUPPI){
```
Comment (slava77) [inline PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h]: please change to ```else if ```

**Comment by slava77** [comment]
@cmsbuild please test

File: PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h (hunk 2 above)
Comment (knash) [reply PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h]: Done.  This was a space v tabs issue

File: PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h (hunk 1 above)
Comment (knash) [reply PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h]: This is irrelevant now, the comments were just used for testing -- taken out

File: PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h (hunk 3 above)
Comment (knash) [reply PhysicsTools/SelectorUtils/interface/PFJetIDSelectionFunctor.h]: done

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/25306	87c06cc
- code changes are in line with the PR description and the follow up review 
    - fixes are provided for existing features in non-standard configurations
    - new  WINTER16 TIGHTLEPVETO ID is introduced
- jenkins tests pass and comparisons with the baseline show no relevant differences

**Comment by fabiocos** [comment]
@peruzzim @fgolf any comment?

**Comment by peruzzim** [comment]
@fabiocos it's just the addition of a working point that was forgotten in the tool, plus a fix for non-standard configuration (expert-only usage).
XPOG is ok with this, should also be ported to 10_2_X and, depending on the interest of the POG to recommend it, to earlier releases we want to support for 2016 analysis.

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Comment by peruzzim** [comment]
could this be backported to 10_2_X please? thanks

**Comment by knash** [comment]
backport to 102X #25608

**Pull request merged**